addons/mail/static/tests/new/composer/composer_tests.js

```diff
@@ -788,10 +787,8 @@ QUnit.test(
             contentType: "text/plain",
             name: "text.txt",
         });
-        await afterNextRender(() =>
-            inputFiles(document.querySelector(".o-mail-composer-core-main .o_input_file"), [file])
-        );
-        await nextTick(); // wait for uploading
+        inputFiles(document.querySelector(".o-mail-composer-core-main .o_input_file"), [file]);
+        await waitUntil(".o-mail-attachment-card");
```

Comment:
Note (I'm writing it on this test but it applies to all of them): there is an important distinction to make depending on whether the attachment is still uploading or already uploaded, the functional aspect of the test is completely different. Not only that, but we had to do a lot of bugfixing in master to fix race conditions related to that.

It would be preferable to always specify that we wait until it's no longer uploading, rather and letting the await pass in either case.